src/Core/Auth/UserFeatures/UserKey/Implementations/RotateUserKeyCommand.cs

```diff
@@ -99,6 +102,17 @@ public async Task<IdentityResult> RotateUserKeyAsync(User user, RotateUserKeyDat
                     _organizationUserRepository.UpdateForKeyRotation(user.Id, model.OrganizationUsers));
             }
 
+            if (model.WebAuthnKeys.Any())
+            {
+                foreach (var webAuthnCredential in model.WebAuthnKeys)
```

Comment:
🎨 : I think this should be changed to use the `delegate` convention as above. It speeds up the database query since it sends a list.

You'll need to update the `ICredentialRepository` to include 
`UpdateEncryptedDataForKeyRotation UpdateForKeyRotation(UserId id, IEnumerable<WebAuthnCredential> credential)`. 

[Interface example](https://github.com/bitwarden/server/blob/main/src/Core/AdminConsole/Repositories/IOrganizationUserRepository.cs#L53)

[Implementation example](https://github.com/bitwarden/server/blob/97b3f3e7eeb0cfb9d94ba04b54b451dac791a8e1/src/Infrastructure.Dapper/AdminConsole/Repositories/OrganizationUserRepository.cs#L541)

CC: @jlf0dev

Reply:
Also, do we in general collect numbers on the performance of specific routes? I'd imagine user key rotation is comparatively rather rare, so it both shouldn't affect the database load much, nor should it be a problem if it takes a few milliseconds more for the user. (I still plan to make this change, trying to establish the process around optimizations though)

Reply:
I think our goal is to use [Open Telemetry ](https://contributing.bitwarden.com/architecture/adr/observability-with-opentelemetry) but I don't know at what level it's implemented.

Reply:
Need to test the queries a bit more on my end, but I think this is implemented correctly now.